process/peer/process.go

```diff
@@ -1126,3 +1163,9 @@ func (vs *validatorStatistics) LastFinalizedRootHash() []byte {
 	defer vs.mutValidatorStatistics.RUnlock()
 	return vs.lastFinalizedRootHash
 }
+
+// EpochConfirmed is called whenever a new epoch is confirmed
+func (vs *validatorStatistics) EpochConfirmed(epoch uint32) {
+	vs.flagSwitchEnabled.Toggle(epoch >= vs.switchEnableEpoch)
```

Comment:
Wondering if there is an chicken-egg problem when using this mechanism in this particular implementation:
Does peer processor is called before or after the epoch change? Because if it is after the epoch change, the new start of epoch block will contain the modified peer changes. Otherwise, it will keep the old processing way. Since the shards do not care about the peer statuses I guess it's ok the processing there.

Reply:
before any processing is called. thus is ok.